[MSE][GStreamer] Skip waiting for preroll when audio sink is non-async

eocanha opened the pull request:
#### 910eddedfb3559fc9ed8829ae84efad7980978a9
<pre>
[MSE][GStreamer] Skip waiting for preroll when audio sink is non-async
<a href="https://bugs.webkit.org/show_bug.cgi?id=262492">https://bugs.webkit.org/show_bug.cgi?id=262492</a>

Reviewed by Alicia Boya Garcia.

When the audio sink is non-async, the GST_STATE_CHANGE_ASYNC change will never happen. Under
these circumstances, the asyncStateChangeDone() code will never run. This patch calls that
method manually, fixing the issue. The method has been now given a better name (didPreroll()),
which better expresses the semantics.

Original author: Pawel Lampe &lt;pawel.lampe@gmail.com&gt;

See: <a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1195">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1195</a>

* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
(WebCore::MediaPlayerPrivateGStreamerMSE::doSeek): Manually call asyncStateChangeDone() when the audio sink is non-async.
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::handleMessage): asyncStateChangeDone() renamed to didPreroll().
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
(WebCore::MediaPlayerPrivateGStreamer::didPreroll): Old asyncStateChangeDone().
(WebCore::MediaPlayerPrivateGStreamer::asyncStateChangeDone): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
(WebCore::MediaPlayerPrivateGStreamerMSE::doSeek): asyncStateChangeDone() renamed to didPreroll().
(WebCore::MediaPlayerPrivateGStreamerMSE::didPreroll): Old asyncStateChangeDone().
(WebCore::MediaPlayerPrivateGStreamerMSE::asyncStateChangeDone): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.h:

Canonical link: <a href="https://commits.webkit.org/268845@main">https://commits.webkit.org/268845@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4ad163fcffa7129a68cde104044f241ed603caa0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20675 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21093 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21748 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22569 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19298 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20912 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24347 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21274 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20629 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20897 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20723 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17978 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23425 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/20862 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17889 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18784 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25047 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/18005 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18968 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18961 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22993 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/20100 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19548 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16611 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/24086 "Built successfully") | 
| [❌ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/20495 "Failed resultsdbpy unit tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18768 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/18620 "Passed tests") | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5739 "Found 3 jsc stress test failures: stress/sampling-profiler-microtasks.js.default, stress/sampling-profiler-microtasks.js.dfg-eager, stress/sampling-profiler-microtasks.js.no-cjit-collect-continuously") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4998 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23105 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/25346 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19366 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5563 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/78e8d686de7ab066830a34660e493e6545fb1839. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/18503

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/02ad7ad7cbd29f63dd7303cdff4b7b724d2c4ac0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/18503

ntrrgc commented on Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
```diff
@@ -191,6 +191,18 @@ bool MediaPlayerPrivateGStreamerMSE::doSeek(const MediaTime& position, float rat
+        if (!audioSinkPerformsAsyncStateChanges) {
+            // If audio-only pipeline's sink is not performing async state changes
+            // we must simulate preroll right away as otherwise nothing will trigger it.
+            asyncStateChangeDone();
```
This makes sense, but we should rename the `asyncStateChangeDone()` to `didPreroll()` so that its name reflects its true purpose. This should be a very simple refactor.

ntrrgc reviewed (changes requested):
Please update the patch with the refactor.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4ad163fcffa7129a68cde104044f241ed603caa0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/18503

ntrrgc reviewed (approved):
LGTM, r+

eocanha committed:
[MSE][GStreamer] Skip waiting for preroll when audio sink is non-async
https://bugs.webkit.org/show_bug.cgi?id=262492

Reviewed by Alicia Boya Garcia.

When the audio sink is non-async, the GST_STATE_CHANGE_ASYNC change will never happen. Under
these circumstances, the asyncStateChangeDone() code will never run. This patch calls that
method manually, fixing the issue. The method has been now given a better name (didPreroll()),
which better expresses the semantics.

Original author: Pawel Lampe <pawel.lampe@gmail.com>

See: https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1195

* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
(WebCore::MediaPlayerPrivateGStreamerMSE::doSeek): Manually call asyncStateChangeDone() when the audio sink is non-async.
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::handleMessage): asyncStateChangeDone() renamed to didPreroll().
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
(WebCore::MediaPlayerPrivateGStreamer::didPreroll): Old asyncStateChangeDone().
(WebCore::MediaPlayerPrivateGStreamer::asyncStateChangeDone): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
(WebCore::MediaPlayerPrivateGStreamerMSE::doSeek): asyncStateChangeDone() renamed to didPreroll().
(WebCore::MediaPlayerPrivateGStreamerMSE::didPreroll): Old asyncStateChangeDone().
(WebCore::MediaPlayerPrivateGStreamerMSE::asyncStateChangeDone): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.h:

Canonical link: https://commits.webkit.org/268845@main

webkit-commit-queue commented:
Committed 268845@main (910eddedfb35): <https://commits.webkit.org/268845@main>

Reviewed commits have been landed. Closing PR #18503 and removing active labels.
